### PR TITLE
Handle binary data already in pbtxt when merging byte-value uploads

### LINKED ISSUE #54: Error with re-saving binary data
Seemingly something to do with our token system for uploads. Steps:
Create a new dataset, attach some non-text binary data to the reaction, and save.
Then refresh the page, attach some more non-text binary data to the reaction, and try to save again.
In the backend is `UnicodeDecodeError`:
``` 
    token = message[:20].decode('utf8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 [or whatever byte] in position 0: invalid start byte
```

### PR DESCRIPTION
**Pull request by n8kim1**
Fixes #54

### CODE REVIEW

**Comment by antonkast-google** [comment]
I don't quite understand what's going on here. What is the contract for these uploads? Should it really be impossible to fail due to invalid encoding?

**Comment by skearnes** [comment]
It seems like this could be a symptom of a larger issue, where we're not properly encoding things on the JS side. See #46 for a change I made on the encoding of enumeration spreadsheets; we probably need something similar for everything that gets sent to the server?

**Comment by n8kim1** [comment]
I think that the bug occurs when `resolve_tokens` recurses through a server-side proto, and checks if any `bytes_values` fields's values are upload tokens (so it can merge the corresponding files). Sometimes, the values are actual binary data that's already been put into the proto (from an earlier merge / upload / save). This seems okay to me, but I think we just check well.

Tokens are specified in `uploads.js` [here](https://github.com/Open-Reaction-Database/ord-editor/blob/ef48632da1538839a4cf7e5b247be0668cdd9304/js/uploads.js#L33). For now, they're always utf-8 decodable. So I think that if a `bytes_values` value is not decodable, then it's not an upload token we're looking for, and we can move on.

I think what's being sent from the JS side is fine -- the `bytes_values` binary data that end up being stored are what we want them to be. (I wonder if we could send them from JS as base64, and decode server-side. I'm not sure that will help this issue, though.)

**Comment by n8kim1** [comment]
TODO -- add a comment in the code, explaining that the form we expect tokens to be in is unicode-decodable

**Comment by n8kim1** [comment]
@antonkast-google @skearnes ready for review! (Also I'll need to format too.)

**Comment by antonkast-google** [comment]
Please ping again after the checks are green.

**Comment by n8kim1** [comment]
@antonkast-google @skearnes checks are passing now, ready for a look.

File: py/serve.py [hunk 1]
```diff
@@ -700,7 +700,16 @@ def resolve_tokens(proto):
...
+                try:
+                    token = message[:20].decode('utf8')
+                except UnicodeDecodeError:
+                    # Client generated tokens are utf-8 decodable.
+                    # (see js/uploads.js) So if the value is not utf-8
+                    # decodable, it's not an upload token, and we should proceed
+                    # to the next bytes_value field. (Likely, the value is
+                    # actual binary data from an upload that's already been
+                    # processed.)
+                    break
```
Comment (skearnes) [inline py/serve.py]: Do you want `continue` here instead of `break`?

Comment (n8kim1) [reply py/serve.py]: Yes -- thanks for the catch!

File: py/serve.py [hunk 2]
```diff
@@ -700,7 +700,16 @@ def resolve_tokens(proto):
...
+                try:
+                    token = message[:20].decode('utf8')
+                except UnicodeDecodeError:
+                    # Client generated tokens are utf-8 decodable.
+                    # (see js/uploads.js) So if the value is not utf-8
+                    # decodable, it's not an upload token, and we should proceed
+                    # to the next proto field. (Likely, the value is
+                    # actual binary data from an upload that's already been
+                    # processed.)
+                    pass
```
Comment (skearnes) [inline py/serve.py]: I think you want `continue` here; `pass` will probably run into errors later on when you try to manipulate the token?

Comment (n8kim1) [reply py/serve.py]: Whoops, that should definitely be continue, I blanked out... Sorry and thanks for catching again.

**Review by skearnes — APPROVED** [review]

**Pull request merged**